src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -1034,1026 +1290,2634 @@ static void Main()
+                      IL_000c:  ldloc.1
+                      IL_000d:  ldloc.2
+                      IL_000e:  ldelem.i4
+                      IL_000f:  stloc.3
+                      IL_0010:  ldloc.0
+                      IL_0011:  ldloc.3
+                      IL_0012:  callvirt   "void System.Collections.Generic.List<int>.Add(int)"
+                      IL_0017:  ldloc.2
+                      IL_0018:  ldc.i4.1
+                      IL_0019:  add
+                      IL_001a:  stloc.2
+                      IL_001b:  ldloc.2
+                      IL_001c:  ldloc.1
+                      IL_001d:  ldlen
+                      IL_001e:  conv.i4
+                      IL_001f:  blt.s      IL_000c
+                      IL_0021:  ldloc.0
+                      IL_0022:  callvirt   "int[] System.Collections.Generic.List<int>.ToArray()"
+                      IL_0027:  ret
+                    }
+                    """,
+                ("ReadOnlySpan<int>", "ReadOnlySpan<int>") =>
+                    // PROTOTYPE: Shouldn't require an intermediate List<int> since the compiler can use e.Length directly.
+                    """
+                    {
+                      // Code size       53 (0x35)
+                      .maxstack  2
+                      .locals init (System.Collections.Generic.List<int> V_0,
+                                    System.ReadOnlySpan<int>.Enumerator V_1,
+                                    int V_2)
+                      IL_0000:  newobj     "System.Collections.Generic.List<int>..ctor()"
+                      IL_0005:  stloc.0
+                      IL_0006:  ldarga.s   V_0
+                      IL_0008:  call       "System.ReadOnlySpan<int>.Enumerator System.ReadOnlySpan<int>.GetEnumerator()"
+                      IL_000d:  stloc.1
+                      IL_000e:  br.s       IL_0020
+                      IL_0010:  ldloca.s   V_1
+                      IL_0012:  call       "ref readonly int System.ReadOnlySpan<int>.Enumerator.Current.get"
+                      IL_0017:  ldind.i4
+                      IL_0018:  stloc.2
+                      IL_0019:  ldloc.0
+                      IL_001a:  ldloc.2
+                      IL_001b:  callvirt   "void System.Collections.Generic.List<int>.Add(int)"
+                      IL_0020:  ldloca.s   V_1
+                      IL_0022:  call       "bool System.ReadOnlySpan<int>.Enumerator.MoveNext()"
+                      IL_0027:  brtrue.s   IL_0010
+                      IL_0029:  ldloc.0
+                      IL_002a:  callvirt   "int[] System.Collections.Generic.List<int>.ToArray()"
+                      IL_002f:  newobj     "System.ReadOnlySpan<int>..ctor(int[])"
+                      IL_0034:  ret
+                    }
+                    """,
+                _ => null
+            };
+            if (expectedIL is { })
+            {
+                verifier.VerifyIL("Program.F", expectedIL);
```

Comment:
I find this layout a little challenging as it "buries the lede" that the thing we're verifying IL of is `Program.F`.

Reply:
With type of thing, I generally make the switch the return of a local function after the call to `VerifyIL` to make it easier to read.